webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/buildHosts/BulkBuildHostModal.js

```diff
@@ -0,0 +1,161 @@
+import { STATUS } from '../../../../constants';
+import { selectAPIStatus } from '../../../../redux/API/APISelectors';
+import { bulkBuildHosts } from './actions';
+
+const BulkBuildHostModal = ({
+  isOpen,
+  closeModal,
+  selectedCount,
+  orgId,
+  fetchBulkParams,
+}) => {
+  const dispatch = useDispatch();
+  const [buildRadioChecked, setBuildRadioChecked] = useState(true);
+  const [rebootChecked, setRebootChecked] = useState(false);
+
+  const hostUpdateStatus = useSelector(state =>
+    selectAPIStatus(state, 'HOST_BUILD_KEY')
+  );
+
+  const handleModalClose = () => {
+    setRebootChecked(false);
+    setBuildRadioChecked(true);
+    closeModal();
+  };
+
+  const handleSave = () => {
+    const requestBody = {
+      included: {
+        search: fetchBulkParams(),
+      },
+      reboot: rebootChecked,
+      rebuild_configuration: !buildRadioChecked,
+    };
+
+    dispatch(bulkBuildHosts(requestBody, handleModalClose, handleModalClose));
+  };
+
+  const handleBuildRadioSelected = selected => {
+    setBuildRadioChecked(selected);
+    if (!selected) {
+      setRebootChecked(false);
+    }
+  };
+
+  const modalActions = [
+    <Button
+      key="add"
+      ouiaId="bulk-build-hosts-modal-add-button"
+      variant="primary"
+      onClick={handleSave}
+      isDisabled={hostUpdateStatus === STATUS.PENDING}
+      isLoading={hostUpdateStatus === STATUS.PENDING}
+    >
+      {__('Confirm')}
+    </Button>,
+    <Button
+      key="cancel"
+      ouiaId="bulk-build-hosts-modal-cancel-button"
+      variant="link"
+      onClick={handleModalClose}
+    >
+      Cancel
+    </Button>,
+  ];
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleModalClose}
+      onEscapePress={handleModalClose}
+      title={__('Build management')}
+      width="50%"
+      position="top"
+      actions={modalActions}
+      id="bulk-build-hosts-modal"
+      key="bulk-build-hosts-modal"
+      ouiaId="bulk-build-hosts-modal"
+    >
+      <TextContent>
+        <Text ouiaId="bulk-set-build-options">
+          <FormattedMessage
+            defaultMessage={__(
+              'Choose an action that will be performed on {hosts}.'
+            )}
+            values={{
+              hosts: (
+                <strong>
+                  <FormattedMessage
+                    defaultMessage="{count, plural, one {# {singular}} other {# {plural}}}"
+                    values={{
+                      count: selectedCount,
+                      singular: __('selected host'),
+                      plural: __('selected hosts'),
+                    }}
+                    id="ccs-options-i18n"
+                  />
+                </strong>
+              ),
+            }}
+            id="bulk-build-host-description"
+          />
+        </Text>
+      </TextContent>
+      <hr />
+      <Radio
+        isChecked={buildRadioChecked}
+        name="buildHostRadioGroup"
+        onChange={() => handleBuildRadioSelected(true)}
```

Comment:
This _may_ just work like this; it's worth a shot. (same for below)

```suggestion
        onChange={handleBuildRadioSelected}
```